controller/monitor/environment_check_monitor.go

```diff
@@ -74,39 +79,44 @@ func NewEnvironmentCheckMonitor(logger logrus.FieldLogger, ds *datastore.DataSto
 		syncCallback: syncCallback,
 	}
 
-	go m.Start()
-
 	return m, nil
 }
 
-func (m *EnvironmentCheckMonitor) Start() {
-	if err := wait.PollUntilContextCancel(m.ctx, m.syncPeriod, true, func(context.Context) (bool, error) {
-		if err := m.run(struct{}{}); err != nil {
-			m.logger.WithError(err).Error("Stopped monitoring environment check")
+func (m *environmentCheckMonitorImpl) Start() {
+	m.startOnce.Do(func() {
+		if err := m.run(); err != nil {
+			m.logger.WithError(err).Error("Initial environment monitoring failed")
 		}
-		return false, nil
-	}); err != nil {
-		if errors.Is(err, context.Canceled) {
-			m.logger.WithError(err).Warning("Environment check monitor is stopped")
-		} else {
-			m.logger.WithError(err).Error("Failed to start environment check monitor")
-		}
-	}
+		go func() {
+			if err := wait.PollUntilContextCancel(m.ctx, m.syncPeriod, false, func(context.Context) (bool, error) {
```

Comment:
Explicitly invoke `m.run` to eliminate the chance that `m.GetCollectedData` get invoked before the first collecting. The `wait.PollUntilContextCancel` is scheduled in another goroutine.